cmd/frontend/internal/auth/userpasswd/handlers.go

```diff
@@ -94,6 +115,22 @@ func handleSignUp(w http.ResponseWriter, r *http.Request, failIfNewUserIsNotInit
 		}
 		newUserData.EmailVerificationCode = code
 	}
+
+	// Prevent abuse (users adding emails of other people whom they want to annoy) with the
+	// following abuse prevention checks.
+	if conf.EmailVerificationRequired() && !newUserData.EmailIsVerified {
+		abused, reason, err := checkEmailAbuse(r.Context(), creds.Email)
+		if err != nil {
+			log15.Error("Error checking email abuse", "email", creds.Email, "error", err)
+			http.Error(w, defaultErrorMessage, http.StatusInternalServerError)
+			return
+		} else if abused {
+			log15.Error("Possible email address abuse prevented", "email", creds.Email, "reason", reason)
+			http.Error(w, "Email address is possibly being abused, please try again later or use a different email address.", http.StatusBadRequest)
```

Comment:
Would 429 be a more appropriate HTTP status code?

```suggestion
			http.Error(w, "Email address is possibly being abused, please try again later or use a different email address.", http.StatusTooManyRequests)
```